torch/lib/c10d/test/ProcessGroupGlooTest.cpp

```diff
@@ -268,87 +269,67 @@ void testSend(const std::string& path) {
   auto tests = CollectiveTest::initialize(path, size);
 
   constexpr uint64_t tag = 0x1337;
-
-  // Create a sender thread and ensure that sendWork can be aborted.
-  std::thread senderThread([&]() {
-    auto selfRank = 0;
-    auto destRank = 1;
-
-    auto& pg = tests[selfRank].getProcessGroup();
-
-    std::vector<at::Tensor> tensors = {
-        at::ones({16, 16}),
-    };
-    // Ensure that we can abort SendWork
-    auto work = pg.send(tensors, destRank, tag);
-    bool sendCompleted;
-    std::thread waitSendWorkThread([&]() { sendCompleted = work->wait(); });
-    work->abort();
-    waitSendWorkThread.join();
-    TORCH_CHECK(!sendCompleted);
-
-    // Ensure that future waits can complete successfully.
-    work = pg.send(tensors, destRank, tag);
-    sendCompleted = work->wait();
-    TORCH_CHECK(sendCompleted);
-  });
-
-  // Helper receiver thread to simulate a real send/recv pair.
-  std::thread receiverThread([&]() {
+  // test that waiting for work to be sent can be aborted successfully.
+  auto selfRank = 0;
+  auto dstRank = 1;
+  std::vector<at::Tensor> tensors = {
+      at::ones({16, 16}),
+  };
+  auto& pg = tests[selfRank].getProcessGroup();
+  auto sendWork = pg.send(tensors, dstRank, tag);
+  bool sendCompleted;
+  std::thread waitSendThreadAbort([&]() { sendCompleted = sendWork->wait(); });
+  sendWork->abort();
+  // Block until the sendWork gets successfully aborted
+  waitSendThreadAbort.join();
+  ASSERT_FALSE(sendCompleted);
+
+  // Now create a separate sender thread to ensure that future waitsends can
```

Comment:
nit: separate receiver thread or does this comment have to be below when we create a sendThread?

Reply:
Right, it should be below